File: Progetto_Finale/Back-End/backend D&D/backend D&D/Models/Entity/Personaggio.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace backend_D_D.Models.Entity
{
    public class Personaggio
    {

        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int PersonaggioID { get; set; }
        [ForeignKey("Utente"), Required]
        public int IdUtente { get; set; }
        [Required, MaxLength(50)]
        public string Nome { get; set; }
        public Utente? Utente { get; set; }
        public ICollection<Armi>? Armi { get; set; }
        public ICollection<Armatura>? Armatura { get; set; }
        public ICollection<Inventario>? Inventario { get; set; }
        public Abilita? Abilita { get; set; }
        public Attributi? Attributi { get; set; }
        public Background? Backgound { get; set; }
        public ICollection<Classi>? Classi { get; set; }
        public Tiri_Salvezza? Tiri_Salvezza { get; set; }
    }
}

File: Progetto_Finale/Front-End/DeD - Copia/src/app/modules/personaggio.ts
import { IAbilita } from "./abilita";
import { IArmatura } from "./armatura";
import { IArmi } from "./armi";
import { IAttributi } from "./attributi";
import { IBackground } from "./background";
import { IClassi } from "./classi";
import { ITiriSalvezza } from "./tiri-salvezza";

export interface IPersonaggio {
    personaggioID: number | null;
    idUtente: number;
    nome: string;
    utente: any | null;
    razza: string;
    moneteOro : number | null;
    armi: IArmi | null;
    armatura: IArmatura[] | null;
    inventario: any | null;
    abilita: IAbilita | null;
    attributi: IAttributi | null;
    backgound: IBackground | null;
    classi: IClassi[] | null,
    tiri_Salvezza: ITiriSalvezza | null;
}
